source/ffgui-window.cpp

```diff
@@ -1350,12 +1376,31 @@ ffguiwin::toggle_custom_resolution()
 void
 ffguiwin::toggle_cropping()
 {
-	int32 state = enablecropping->Value();
 
-	topcrop->SetEnabled(state);
-	bottomcrop->SetEnabled(state);
-	leftcrop->SetEnabled(state);
-	rightcrop->SetEnabled(state);
+	if (outputvideoformatpopup->FindMarkedIndex() == 0)
+	{
+		enablecropping->SetEnabled(false);
+	}
+	else
+	{
+		enablecropping->SetEnabled(true);
+	}
+
+	bool options_enabled;
```

Comment:
From here on we could leave the:

```
	int32 state = enablecropping->Value();

 	topcrop->SetEnabled(state);
 	bottomcrop->SetEnabled(state);
 	leftcrop->SetEnabled(state);
 	rightcrop->SetEnabled(state);
```

Reply:
No, actually not ;-). The Value() method gives you the information whether the checkbox is checked or not. But we also need to know if it is disabled or not. (it can be checked, but disabled). It worked as long as the enablecropping checkbox was never disabled, but even then it was semantically a bit ugly (sorry ;-)) because you passed the int32 state flag into SetEnabled() which takes a bool. It worked correctly because B_CONTROL_ON is non-zero and therefore evaluates to true, but I'd rather not rely on that.

Reply:
> because you passed the int32 state flag into SetEnabled() which takes a bool.

I don't expect B_CONTROL_ON to ever change. Would casting it to bool be nicer?
Anyway, you surely know better than I... Just go ahead. :)

Reply:
Or, better: `bool state = (enablecropping->Value() == B_CONTROL_ON);`
That should be a proper bool. :)

Reply:
Ok, I'm all for good humor in coding (and all other aspects of life, for that matter ;-) ), but you've lost me with the last one. Let's keep it simple and readable, so that even newbies can have fun and join us in coding. Kind of an antidote to the famous Linus Torvalds attitude (though I admire him for some things, technical elitism is not one of them)
After all the compiler optimizes away the differences anyway.

Reply:
Hmmm... I wouldn't consider this unreadable at all...
You compare `enablecropping->Value()` with `B_CONTROL_ON` and store the result (true or false) as a bool.

Reply:
Not unreadable as far as the code is concerned. I just find my solution easier to read in the sense that it makes clear that we are checking for 2 different things, the enabled/disabled and the checked/unchecked state of the control. 
But that's just the way I see it, that doesn't make your solution any less valid.